docs/book/v3/features/container/config.md

```diff
@@ -0,0 +1,213 @@
+# Container configuration
+
+> This chapter is primarily written for container providers, so that they know
+> what configuration features must be compatible, and what compatibility
+> ultimately means within the project.
+
+[PSR-11](https://www.php-fig.org/psr/psr-11/) defines an interface for
+dependency injection containers, and that interface is geared towards
+_consumption_ of the container &mdash; not _population_ of it.
+
+Expressive _consumes_ a PSR-11 container, but also provides _configuration_ for
+a container: it defines what services it needs, and how to create them.
+
+As such, any container consumed by Expressive must also understand its
+configuration format, and deliver consistent understanding of that format when
+providing services based on it.
+
+This document describes the configuration format, and details expectations for
+implementations.
+
+## The format
+
+Container configuration is provided within the `dependencies` key of
+configuration. That key is structured as follows:
+
+```
+return [
+    'dependencies' => [
+        'services' => [
+            // name => instance pairs
+        ],
+        'aliases' => [
+            // alias => target pairs
+        ],
+        'factories' => [
+            // service => factory pairs
+        ],
+        'invokables' => [
+            // service => instantiable class pairs
+        ],
+        'delegators' => [
+            // service => array of delegator factories
+        ],
+    ],
+];
+```
+
+## Services
+
+_Services_ are actual instances you want to retrieve later from the container.
+These are generally provided at initial creation; the `config` service is
+populated in this way.
+
+When retrieving a service mapped in this way, you will always receive the
+initial instance.
+
+## Aliases
+
+_Aliases_ map a service _alias_ to another service, and are provided as
+key/value pairs. As an example:
+
+```php
+'aliases' => [
+    'Zend\Expressive\Delegate\DefaultDelegate' => \Zend\Expressive\Handler\NotFoundHandler::class,
+],
+```
+
+In this case, if the service named "Zend\\Expressive\\Delegate\\DefaultDelegate"
+is requested, the container should resolve that to the service
+`Zend\Expressive\Handler\NotFoundHandler` and return that instead.
+
+When returning an aliased service, the container MUST return the same instance
+as if the target service were retrieved.
+
+Aliases may reference other aliases, as well. In such cases, the above rules
+apply to the final resolved service, and not any intermediary aliases.
```

Comment:
I'm updating zend-container-test library and still for me it's not clear for what I can define aliases. I mean if it is possible to define here alias which reference to:
- service
- invokable
- factory
- delegator
- alias
?